### PR TITLE
added makefile

### PR DESCRIPTION
**Pull request by dutterbutter**
**Changes made in this pull request:**

- added Makefile with the following commands:
        - **make help**: this will outline all possible commands within makefile
        - **make lint** : this uses `gometalinter` and will `go get` the package if it's not present
        - **make test** : this will run the `go test ./...` command
        - **make install** : this will install dependencies
        - **make clean** : this will clean build cache

Since we don't have any go files present, these will return an error.

The rationale for the makefile is to be able to use common commands with general ease. Moving forward the Makefile will be primarily used for different builds much like its used in `go-ethereum`.

 [Here](https://azer.bike/journal/a-good-makefile-for-go/) is a good link for an overview of a makefile in go

### CODE REVIEW

**Comment by dutterbutter** [comment]
@priom avoid merging `development` into PR branches when 2 parties have approved the PR branch will be merged into `development`. If there are conflicts, ask the individual to resolve the conflicts.

**Comment by dutterbutter** [comment]
@ansermino added `make help` as requested, this command will show outline of possible makefile commands

**Comment by ansermino** [comment]
@dutterbutter `make` defaults to `make test`, please change it to `make help`

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	go test $(PKGS)
+
+$(GOMETALINTER):
+	go get -u github.com/alecthomas/gometalinter
+	gometalinter --install &> /dev/null
+
+## lint: Lints project files, go gets gometalinter if missing. Runs `gometalinter` on project files.
+.PHONY: lint
+lint: $(GOMETALINTER)
+	gometalinter ./... exclude=gosec --vendor
```
Comment (ansermino) [inline Makefile]: Remove exclusion

**Comment by dutterbutter** [comment]
@ansermino requested changes made

**Pull request merged**